charts/fluentd/templates/hpa.yaml

```diff
@@ -1,5 +1,5 @@
 {{- if and  ( eq  .Values.kind "Deployment" )  .Values.autoscaling.enabled  }}
-apiVersion: autoscaling/v2beta2
+apiVersion: {{ include "fluentd.hpa.apiVersion" . }}
```

Comment:
Could we move the hpa from helper template into here?

There is no really usefull reason why you need to put the fluentd.hpa.apiVersion in the _helpers.tpl as this is only used for HPA and the helpers template is a generic template.

Reply:
Would have been better if kept under hpa.yaml. However, don't really see where we have defined that only generic definitions as be part of _helpers.tpl and not helpers which are needed different yaml. 

I see this pattern in other open source code as well. Common definitions at single place - https://github.com/bitnami/charts/blob/main/bitnami/common/templates/_capabilities.tpl

@dioguerra - we need this piece of code. Would help if we can resolve this.

Reply:
Requesting more eyeballs @stevehipwell

Reply:
Can we merge this PR if this looks good ?